### ISSUE TITLE
Support version in workspace inheritance

### ISSUE THREAD

**Issue by leighmcculloch** [pksunkara/cargo-workspaces#82]
With workspace inheritance released in Rust 1.64 the version of packages can be specified in the workspace Cargo.toml file.

E.g.:
```toml
[workspace]

[workspace.package]
version = "1.0.0"
```

And then the crate Cargo.toml can reference it.

E.g.:
```toml
[package]
version.workspace = true
```

Currently cargo-workspaces can read the versions of crates that use inheritance for their version, but when setting or bumping a version, it will not update the workspace version.

**Comment by pksunkara** [comment]
Thanks for the report. (Hopefully you give the example completely). I will look into fixing this during the weekend.

**Comment by leighmcculloch** [comment]
He's an example:

```
❯ git clone https://github.com/stellar/rs-soroban-env

❯ git checkout 99de1bf0d2026d1fdb2854a13c195c5902f088a4

❯ cd rs-soroban-env

❯ cargo workspaces version --all --force '*' --no-git-commit --yes custom 0.0.10
info looking for changes since v0.0.9
info current common version 0.0.9

Changes:
 - soroban-env-common: 0.0.9 => 0.0.10
 - soroban-env-guest: 0.0.9 => 0.0.10
 - soroban-env-host: 0.0.9 => 0.0.10
 - soroban-env-macros: 0.0.9 => 0.0.10
 - soroban-native-sdk-macros: 0.0.9 => 0.0.10
 - soroban-synth-wasm: 0.0.9 => 0.0.10
 - soroban-test-wasms: 0.0.9 => 0.0.10
 - test_no_std: 0.0.9 => 0.0.10

    Updating crates.io index
    Updating git repository `https://github.com/stellar/wasmi`
    Updating git repository `https://github.com/stellar/rs-stellar-xdr`
    Blocking waiting for file lock on package cache
    Updating crates.io index
    Updating git repository `https://github.com/stellar/wasmi`
    Updating git repository `https://github.com/stellar/rs-stellar-xdr`
    Updating crates.io index
    Updating git repository `https://github.com/stellar/wasmi`
    Updating git repository `https://github.com/stellar/rs-stellar-xdr`
    Updating crates.io index
    Updating git repository `https://github.com/stellar/rs-stellar-xdr`
    Updating crates.io index
    Updating crates.io index
    Updating git repository `https://github.com/stellar/wasmi`
    Updating git repository `https://github.com/stellar/rs-stellar-xdr`
    Updating crates.io index
    Updating git repository `https://github.com/stellar/wasmi`
    Updating git repository `https://github.com/stellar/rs-stellar-xdr`
info success ok

❯ git status
HEAD detached at 99de1bf
nothing to commit, working tree clean
```

**Comment by pksunkara** [comment]
Released 0.2.38
